handlers/announcement.go

```diff
@@ -118,73 +123,121 @@ func parseSpan(a *html.Node) (text string, link string) {
-			for _, node := range item.Nodes {
-				for n := node.FirstChild; n != nil; n = n.NextSibling {
-					if n.Type != html.ElementNode {
-						continue
-					}
-					if n.Data != "a" && n.Data != "span" {
-						continue
-					}
+	// Find the announcements
+	doc.Find("div.comman-inner-section").Find("p").Each(func(i int, item *goquery.Selection) {
+		for _, node := range item.Nodes {
+			for n := node.FirstChild; n != nil; n = n.NextSibling {
+				if n.Type != html.ElementNode {
+					continue
+				}
+				if n.Data != "a" && n.Data != "span" {
+					continue
+				}
 
-					// Loop to previous siblings until a text node is found
-					var date, PrevSibling string
-					for prev := n.PrevSibling; prev != nil; prev = prev.PrevSibling {
-						if prev.Data == "span" {
-							PrevSibling = getTextInSpan(prev)
-						} else if prev.Type == html.TextNode {
-							PrevSibling = prev.Data
-						}
-
-						if PrevSibling != "" {
-							break
-						}
+				// Loop to previous siblings until a text node is found
+				var date, PrevSibling string
+				for prev := n.PrevSibling; prev != nil; prev = prev.PrevSibling {
+					if prev.Data == "span" {
+						PrevSibling = getTextInSpan(prev)
+					} else if prev.Type == html.TextNode {
+						PrevSibling = prev.Data
 					}
-					date = strings.TrimSpace(PrevSibling)
 
-					var title, link string
-					if n.Data == "a" {
-						title, link = parseA(n)
-					} else if n.Data == "span" {
-						title, link = parseSpan(n)
+					if PrevSibling != "" {
+						break
 					}
+				}
+				date = strings.TrimSpace(PrevSibling)
 
-					tags := fetchTags(title)
-					if title != "" && link != "" {
-						announcements = append(announcements, Announcement{Date: date, Title: title, Link: link, Tags: tags})
-					}
+				var title, link string
+				if n.Data == "a" {
+					title, link = parseA(n)
+				} else if n.Data == "span" {
+					title, link = parseSpan(n)
+				}
+
+				tags := fetchTags(title)
+				if title != "" && link != "" {
+					announcements = append(announcements, Announcement{Date: date, Title: title, Link: link, Tags: tags})
 				}
 			}
-		})
+		}
+	})
+
+	return announcements
+}
+
+// insertNewAnnouncements saves announcements to database
+//
+// It first calls scrapeAnnouncements() and then saves the results after
+// some formatting
+func insertNewAnnouncements(db *sql.DB) {
+    queries := query.New(db)
+	ctx := context.Background()
+	var announcements = scrapeAnnouncements()
+    for i := range announcements {
+        announcements[i].Date =  strings.Replace(announcements[i].Date,".","-",3)
+        announcements[i].Date = strings.Replace(announcements[i].Date,"/","-",3)
+        announcements[i].Date = strings.Replace(announcements[i].Date,"__","",2)
+        if announcements[i].Date == "1-12-2021" {
+            announcements[i].Date = "01-12-2021"
+        }
+        if announcements[i].Date == "07-05-019" {
+            announcements[i].Date = "07-05-2019"
+        }
```

Comment:
extract into a function.
maybe just into scrape function.

Reply:
Good idea, will create a dateHandler() function of some sort to keep all such cases